cmd/account/iam-cred-rotate.go

```diff
@@ -0,0 +1,1578 @@
+	if err != nil {
+		o.log.Warn(o.ctx, "fetchCredentialsRequests returned err:'%s'\n", err)
+		return err
+	}
+	fmt.Printf("\nOptions --describe-keys, --describe-secrets can be used to provide additional info/status\n")
+	fmt.Printf("Script Run Completed Successfully.\n")
+	return nil
+}
+
+func (o *rotateCredOptions) printJsonYaml(obj any) error {
+	if o.output == jsonFormat {
+		pbuf, err := json.MarshalIndent(obj, "", "    ")
+		if err != nil {
+			o.log.Error(o.ctx, "Failed to marshal json. Err:'%v'\n", err)
+			return err
+		}
+		fmt.Printf("%s\n", string(pbuf))
+		return nil
+	}
+	if o.output == yamlFormat {
+		pbuf, err := yaml.Marshal(obj)
+		if err != nil {
+			o.log.Error(o.ctx, "Failed to marshal yaml. Err:'%v'\n", err)
+			return err
+		}
+		fmt.Printf("%s\n", string(pbuf))
+		return nil
+	}
+	return fmt.Errorf("printJsonYaml function called while format set to:'%s'", o.output)
+}
+
+// Some very basic 'early' validation on the key input if provided.
+// AWS docs, access key constraints: Minimum length of 16. Maximum length of 128.
+func (o *rotateCredOptions) isValidAccessKeyId(keyId string) bool {
+	if len(keyId) < 16 || len(keyId) > 128 {
+		o.log.Error(o.ctx, "Invalid Access key length:%d\n", len(keyId))
+		return false
+	}
+	return true
+}
+
+func (o *rotateCredOptions) printClusterInfo() error {
+	// currently using account byoc value for ccs sanity checks, is this needed?
+	isCCS, err := utils.IsClusterCCS(o.ocmConn, o.clusterID)
+	if err != nil {
+		return err
+	}
+	o.log.Debug(o.ctx, "Is Cluster CCS?:%t\n", isCCS)
+	output := os.Stdout
+	if o.output != standardFormat {
+		output = os.Stderr
+	}
+	w := tabwriter.NewWriter(output, 1, 1, 1, ' ', 0)
+	fmt.Fprintf(w, "\n----------------------------------------------------------------------\n")
+	fmt.Fprintf(w, "Cluster ID:\t%s\n", o.clusterID)
+	fmt.Fprintf(w, "Cluster External ID:\t%s\n", o.cluster.ExternalID())
+	fmt.Fprintf(w, "Cluster Name:\t%s\n", o.cluster.Name())
+	fmt.Fprintf(w, "Cluster Is CCS:\t%t\n", isCCS)
+	fmt.Fprintf(w, "----------------------------------------------------------------------\n")
+	w.Flush()
+	return nil
+}
+
+// Get list of credentialsRequests with openshift namespaces for AWS provider type.
+// The secrets referenced by these credReqs will need to be updated after cred rotation.
+func (o *rotateCredOptions) getAWSCredentialsRequests(nameSpace string) ([]credreqv1.CredentialsRequest, error) {
+	// Make sure we have a backplane connection to the target cluster...
+	if o.clusterKubeClient == nil {
+		err := o.connectClusterClient()
+		if err != nil {
+			return nil, err
+		}
+	}
+	if o.ctx == nil {
+		return nil, fmt.Errorf("invalid ctx, nil")
+	}
+	const AWSProviderSpecType string = "AWSProviderSpec"
+	delCredReqList := []credreqv1.CredentialsRequest{}
+	var credRequestList credreqv1.CredentialsRequestList
+	var listOptions = client.ListOptions{}
+	// TODO:!! If 'nameSpace' is not provided query is agaist all namespaces.
```

Comment:
I'm seeing same.  1 CR from "openshift-cloud-ingress-operator" and 2 from  "openshift-velero"...
```
./osdctl account iam-secret-mgmt -S ${CLUSTER_ID} --reason "${REASON}" --describe-secrets -o json | jq '.[] | .CredentialRequestNamespace'

"openshift-cloud-credential-operator"
"openshift-cloud-credential-operator"
"openshift-cloud-credential-operator"
"openshift-cloud-credential-operator"
"openshift-cloud-credential-operator"
"openshift-cloud-credential-operator"
"openshift-cloud-ingress-operator"
"openshift-velero"
"openshift-velero"

```

Reply:
Thanks @clcollins,  definitely would be nice to provide a list of namespaces to the request here :-). 
I can look into reversing this to instead first filter out "openshift-*" namespaces, and then fetch the cred requests from each openshift namespace.  Today we have ~100 openshift namespaces, so this would add ~100 'credentialRequestList' request()s to the util, but agree this might be less intrusive from a customer data perspective.